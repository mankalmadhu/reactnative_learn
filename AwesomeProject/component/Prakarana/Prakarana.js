
const Prakarana = (props) => (
  <View style={prakaranaVinyasa.container}>
    <Image source={{ uri: props.picture.large}} style={prakaranaVinyasa.photo} />
    <Text style={prakaranaVinyasa.text}>
      {`${props.name.first} ${props.name.last}`}
    </Text>
  </View>
);

export default Prakarana;