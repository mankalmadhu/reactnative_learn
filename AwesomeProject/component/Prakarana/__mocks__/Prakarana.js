import React from 'react';

const Prakarana = (props) => (
    <div>
      Prakarana {`${props.name.first} ${props.name.last}`}
    </div>
   
);

export default Prakarana;