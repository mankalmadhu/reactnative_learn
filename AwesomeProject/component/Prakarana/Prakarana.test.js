import 'react-native';
import React from 'react';
import Prakarana from './Prakarana';

import Renderer from 'react-test-renderer';

const testData = {"picture":{"large":"largepic"}, "name":{"first":"first","last":"last"}};

test("Prakarana Render" , () => {

   const prakaranaRenderer = Renderer.create(
       <Prakarana {...testData}/>
   ).toJSON();

   expect(prakaranaRenderer).toMatchSnapshot();

});