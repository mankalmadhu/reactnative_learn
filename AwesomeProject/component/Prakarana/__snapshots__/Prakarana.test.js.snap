// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prakarana Render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "padding": 12,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "largepic",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
        "marginLeft": 12,
      }
    }
  >
    first last
  </Text>
</View>
`;
